### ISSUE TITLE
Error sharing local portfolio

### ISSUE THREAD

**Issue by jordan-evans** [awslabs/aws-service-catalog-puppet#89]
Getting an error when sharing a portfolio locally using `0.3.0`. Snippet is below

```
INFO MainThread [test-base] 012345678910:eu-west-2 :: test-base-audit-compliance status: FAILED 
ERROR: [pid 329] Worker Worker(salt=011027649, workers=10, host=768f187a1b2a, username=root, pid=100) failed    ImportIntoSpokeLocalPortfolioTask(account_id=012345678910, region=eu-west-2, portfolio=test-base, hub_portfolio_id=port-gwr7wrqzepuny) 
Traceback (most recent call last): 
  File "/usr/local/lib/python3.7/site-packages/luigi/worker.py", line 199, in run 
    new_deps = self._run_get_new_deps() 
  File "/usr/local/lib/python3.7/site-packages/luigi/worker.py", line 139, in _run_get_new_deps 
    task_gen = self.task.run() 
  File "/usr/local/lib/python3.7/site-packages/servicecatalog_puppet/luigi_tasks_and_targets.py", line 525, in run 
    raise Exception(f"[{self.portfolio}] {self.account_id}:{self.region} :: Copying " 
Exception: [test-base] 012345678910:eu-west-2 :: Copying test-base-audit-compliance failed: Internal Failure 
ERROR MainThread [pid 329] Worker Worker(salt=011027649, workers=10, host=768f187a1b2a, username=root, pid=100) failed    ImportIntoSpokeLocalPortfolioTask(account_id=012345678910, region=eu-west-2, portfolio=test-base, hub_portfolio_id=port-gwr7wrqzepuny) 
Traceback (most recent call last): 
  File "/usr/local/lib/python3.7/site-packages/luigi/worker.py", line 199, in run 
    new_deps = self._run_get_new_deps() 
  File "/usr/local/lib/python3.7/site-packages/luigi/worker.py", line 139, in _run_get_new_deps 
    task_gen = self.task.run() 
  File "/usr/local/lib/python3.7/site-packages/servicecatalog_puppet/luigi_tasks_and_targets.py", line 525, in run 
    raise Exception(f"[{self.portfolio}] {self.account_id}:{self.region} :: Copying " 
Exception: [test-base] 012345678910:eu-west-2 :: Copying test-base-audit-compliance failed: Internal Failure 
INFO: Informed scheduler that task   ImportIntoSpokeLocalPortfolioTask_012345678910_port_gwr7wrqzepu_test_base_c1f5fe62bc   has status   FAILED 
INFO MainThread Informed scheduler that task   ImportIntoSpokeLocalPortfolioTask_012345678910_port_gwr7wrqzepu_test_base_c1f5fe62bc   has status   FAILED 
```

Manifest snippet:

```
spoke-local-portfolios:
  optional-products-for-spokes:
    portfolio: test-base
    depends_on: []
    associations:
      - arn:aws:iam::${AWS::AccountId}:role/ServiceCatalogDeploymentRole
      - arn:aws:iam::${AWS::AccountId}:role/servicecatalog-puppet/PuppetRole
    constraints:
      launch:
        - products: "test-*"
          roles:
            - arn:aws:iam::${AWS::AccountId}:role/ServiceCatalogDeploymentRole
    deploy_to:
      tags:
        - tag: scope:spoke
          regions: default_region
```

Portfolio is created in spoke with associations, but no products have been copied across.

**Comment by eamonnfaherty** [comment]
Could I please see the rest of your manifest.

**Comment by jordan-evans** [comment]
Turns out this is because the sharing portfolio (i.e `test-base`) wasn't imported into the spoke account. Looks like that is required for the products to be copied to the local portfolio.

Suggest that targeting a portfolio using `spoke-local-portfolios` automatically shares the portfolio with target accounts maybe? Workaround for this for me was to use a kind of dummy launch to deploy a product to that account (which forced Puppet to to the sharing).

**Comment by eamonnfaherty** [comment]
this is resolved in 0.4.0
